### PR TITLE
[master < T1266] Improve deserialization performance

### PR DESCRIPTION
**Pull request by as51340**
## Preparation query:
```cypher
UNWIND range(1, 100000) as u CREATE (n:Node {string: "test"}) RETURN n;
```
## Before optimization:
```cypher
MATCH (n) RETURN n;
```
**Ran 5 times, avg. is: 0.5718**

## After:
```cypher
MATCH (n) RETURN n;
```

**Ran 5 times, avg. is: 0.3094**

### Conclusion: Roughly this PR gives 2x faster data deserialization from disk. Loading all edges isn't improved because the cost comes from different place.

[master < Epic] PR
- [ ] Check, and update documentation if necessary
- [ ] Write E2E tests
- [ ] Compare the [benchmarking results](https://bench-graph.memgraph.com/) between the master branch and the Epic branch
- [ ] Provide the full content or a guide for the final git message

[master < Task] PR
- [ ] Check, and update documentation if necessary
- [ ] Provide the full content or a guide for the final git message

To keep docs changelog up to date, one more thing to do:
- [ ] Write a release note here, including added/changed clauses
- [ ] Tag someone from docs team in the comments

### CODE REVIEW

File: src/storage/v2/id_types.hpp [hunk 1]
```diff
@@ -31,6 +32,11 @@ namespace memgraph::storage {
     static name FromInt(int64_t id) { return name{utils::MemcpyCast<uint64_t>(id)}; }                         \
     uint64_t AsUint() const { return id_; }                                                                   \
     int64_t AsInt() const { return utils::MemcpyCast<int64_t>(id_); }                                         \
+    static name FromString(std::string_view id) {                                                             \
+      uint64_t value;                                                                                         \
+      std::from_chars(id.data(), id.data() + id.size(), value);                                               \
```
Comment (Darych) [inline src/storage/v2/id_types.hpp:37]: What do you think about checking return values of `from_chars`?
I wrote checks in https://github.com/memgraph/memgraph/blob/master/src/utils/rocksdb_serialization.hpp#L214
I think there we also need to change to `FromString()` function call

File: src/storage/v2/mvcc.hpp [hunk 2]
```diff
@@ -131,13 +131,17 @@ inline Delta *CreateDeleteDeserializedObjectDelta(Transaction *transaction, std:
                                                   std::string &&ts) {
   transaction->EnsureCommitTimestampExists();
   // Should use utils::DecodeFixed64(ts.c_str()) once we will move to RocksDB real timestamps
-  return &transaction->deltas.use().emplace_back(Delta::DeleteDeserializedObjectTag(), std::stoull(ts), old_disk_key);
+  uint64_t ts_id = 0;
+  std::from_chars(ts.data(), ts.data() + ts.size(), ts_id);
```
Comment (Darych) [inline src/storage/v2/mvcc.hpp:135]: Maybe create utils function which will convert string_view to uint64_t?
We will be able to use it here and in LabelId also.

File: src/storage/v2/id_types.hpp (hunk 1 above)
Comment (as51340) [reply src/storage/v2/id_types.hpp:37]: The reason I went without check is the performance. Check will obviously slow down things and I think that if something bad happens, std::from_chars will throw exception automatically.

File: src/storage/v2/mvcc.hpp (hunk 2 above)
Comment (as51340) [reply src/storage/v2/mvcc.hpp:135]: I can add this

File: src/storage/v2/disk/name_id_mapper.hpp [hunk 3]
```diff
@@ -44,7 +44,8 @@ class DiskNameIdMapper final : public NameIdMapper {
     }
     uint64_t res_id = 0;
     if (auto maybe_id_from_disk = name_to_id_storage_->Get(std::string(name)); maybe_id_from_disk.has_value()) {
-      std::from_chars(maybe_id_from_disk->data(), maybe_id_from_disk->data() + maybe_id_from_disk->size(), res_id);
+      auto id_disk_value = maybe_id_from_disk.value();
+      std::from_chars(id_disk_value.data(), id_disk_value.data() + id_disk_value.size(), res_id);
```
Comment (Darych) [inline src/storage/v2/disk/name_id_mapper.hpp:48]: Can we call here `utils::ParseStringToUint64`?

**Review by Darych — APPROVED** [review]
LGTM

**Comment by as51340** [comment]
Release note: The reading performance of vertices has been improved 2x.

**Pull request merged**